DuckDuckGo/SyncSettingsViewController.swift

```diff
@@ -333,15 +163,17 @@ extension SyncSettingsViewController: ScanOrPasteCodeViewModelDelegate {
                 showDeviceConnected()
                 return true
             } else if let connectKey = syncCode.connect {
+                if syncService.account == nil {
+                    try await syncService.createAccount(deviceName: deviceName, deviceType: deviceType)
+                    rootView.model.syncEnabled(recoveryCode: recoveryCode)
+                }
                 try await syncService.transmitRecoveryKey(connectKey)
                 dismissPresentedViewController()
                 return true
             }
 
         } catch {
-            if !(error is SyncError) {
-                assertionFailure(error.localizedDescription)
-            }
+            handleError(error)
```

Comment:
When trying testing steps from the PR description I'm sometimes landing here with “accountAlreadyExists” error. I’m unable to debug it, because with the debugger it always seems to be working fine 🤯. It also works fine on develop.

Reply:
Yeah working fine on simulator, but I actually managed to get this while debugging on the iPad so hopefully can fix.

Reply:
It's the scanner. I think it's sending / processing too many camera events.

Reply:
I've updated the code to use an actor and be more robust about killing the scanner once it detects a code, see if that helps.

Reply:
Nice catch! And that explains why it worked fine with some breakpoints set up :)